hypex/ab_test/aa_tester.py

```diff
@@ -889,6 +762,176 @@ def cat_feature_uniform_analysis(
         figure.suptitle(f"{control_data.name}", fontsize=kwargs.get("title_size", 20))
         plt.show()
 
+    def __get_test_and_control_series(
+        self,
+        test_group: pd.Series = None,
+        control_group: pd.Series = None,
+        data: pd.DataFrame = None,
+        target_field: str = None,
+    ):
+        if test_group is None and control_group is None:
+            if data is None or target_field is None:
+                raise ValueError(
+                    "test_group and control_group cannot be None if data and target_field are None"
+                )
+            splited_data = self.split_splited_data(data)
+            test_group = splited_data["test"][target_field]
+            control_group = splited_data["control"][target_field]
+        return {"test_group": test_group, "control_group": control_group}
+
+    def calc_mde(
+        self,
+        test_group: pd.Series = None,
+        control_group: pd.Series = None,
+        data: pd.DataFrame = None,
+        target_field: str = None,
+        reliability: float = 0.95,
+        power: float = 0.8,
+    ) -> float:
+        """Calculates the minimum detectable effect (MDE) for a given test and control groups.
+
+        Args:
+            test_group:
+                The test group as a pandas Series
+            control_group:
+                The control group as a pandas Series
+            data:
+                The input data as a pandas DataFrame used if test_group and control_group are None
+            target_field:
+                The target field used if given data is a DataFrame
+            reliability:
+                The reliability of the test
+            power:
+                The power of the test
+
+        Returns:
+            The minimum detectable effect
+        """
+
+        m = norm.ppf(1 - (1 - reliability) / 2) + norm.ppf(power)
+
+        groups = self.__get_test_and_control_series(
+            test_group=test_group,
+            control_group=control_group,
+            data=data,
+            target_field=target_field,
+        )
+        test_group = groups["test_group"]
+        control_group = groups["control_group"]
+
+        n_test, n_control = len(test_group), len(control_group)
+        proportion = n_test / (n_test + n_control)
+        p = np.sqrt(1 / (proportion * (1 - proportion)))
+
+        var_test, var_control = np.var(test_group, ddof=1), np.var(
+            control_group, ddof=1
+        )
+        s = np.sqrt(var_test / n_test + var_control / n_control)
+
+        return p * m * s
+
+    def calc_sample_size(
+        self,
+        test_group: pd.Series = None,
+        control_group: pd.Series = None,
+        data: pd.DataFrame = None,
+        target_field: str = None,
+        group_col: str = "group",
```

Comment:
No usage of this variable in functions' body

Reply:
Yes. This is bug. I will fix it.